integ/components/deadline/common/scripts/bash/component_e2e.sh

```diff
@@ -19,12 +27,15 @@ if [[ ! "${SKIP_TEST_CHECK}" = "true" ]]; then
 
   # Load utility functions
   source "../common/scripts/bash/deploy-utils.sh"
-    
+
   if [[ $OPTION != '--destroy-only' ]]; then
+    run_hook
     deploy_component_stacks $COMPONENT_NAME
+    run_hook
     execute_component_test $COMPONENT_NAME
   fi
   if [[ $OPTION != '--deploy-and-test-only' ]]; then
+    run_hook
```

Comment:
I don't see a need for a separate hook for each phase so I'll change the name as you suggested. I'm thinking about calling it `PRE_AWS_INTERACTION_HOOK` since it is meant to run before any interaction with AWS.

Reply:
Sounds good. Let's go with this